src/ORM/DB.php

```diff
@@ -696,4 +704,100 @@ public static function alteration_message($message, $type = "")
     {
         self::get_schema()->alterationMessage($message, $type);
     }
+
+    /**
+     * Check if all tables and field columns for a class exist in the database.
+     *
+     * @param string $class
+     * @return boolean
+     */
+    public static function databaseIsReady(string $class): bool
+    {
+        if (!is_subclass_of($class, DataObject::class)) {
+            throw new InvalidArgumentException("$class is not a subclass of " . DataObject::class);
+        }
+
+        // Don't check again if we already know the db is ready for this class.
+        // Necessary here before the loop to catch situations where a subclass
+        // is forced as ready without having to check all the superclasses.
+        if (!empty(self::$databaseReadyClasses[$class])) {
+            return true;
+        }
+
+        // Check if all tables and fields required for the class exist in the database.
+        $requiredClasses = ClassInfo::dataClassesFor($class);
```

Comment:
@emteknetnz Technically this only actually needs to check for ancestors (if a subclass isn't ready it doesn't impact on the current class's readiness), but is currently checking for the full hierarchy. I think it makes things simpler in the long run but if some project has a lot of subclasses of whatever it is they're actually checking it could potentially result in a performance hit. I don't think it's anything to worry about personally but just wanna check with you.